core/src/mindustry/world/blocks/defense/MendProjector.java

```diff
@@ -92,9 +120,36 @@ public void updateTile(){
         public void drawSelect(){
             float realRange = range + phaseHeat * phaseRangeBoost;
 
-            indexer.eachBlock(this, realRange, other -> true, other -> Drawf.selected(other, Tmp.c1.set(baseColor).a(Mathf.absin(4f, 1f))));
+            indexer.eachBlock(this, realRange, other -> true, other -> Drawf.selected(other, baseColor.cpy().lerp(phaseColor, phaseHeat).a(Mathf.absin(4f, 1f))));
 
-            Drawf.dashCircle(x, y, realRange, baseColor);
+            Drawf.dashCircle(x, y, realRange, baseColor.cpy().lerp(phaseColor, phaseHeat));
+            
+            if(!cons().optionalValid() && hasBoost && boosterUnlocked()){
+                indexer.eachBlock(this, range + phaseRangeBoost, other -> true, other -> {
+                  if(!Mathf.within(x, y, other.x, other.y, range)){
+                      Drawf.selected(other, phaseColor.cpy().a(Mathf.absin(4f, 1f)));
+                  }
```

Comment:
Tab size should always be 4 spaces.
```suggestion
                    if(!Mathf.within(x, y, other.x, other.y, range)){
                        Drawf.selected(other, phaseColor.cpy().a(Mathf.absin(4f, 1f)));
                    }
```